src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift

```diff
@@ -1749,4 +1749,53 @@ class HealthCertificateServiceTests: CWATestCase {
 		XCTAssertEqual(person.healthCertificates[0].vaccinationEntry?.uniqueCertificateIdentifier, "newCertificate")
 		XCTAssertEqual(store.recycleBinItems.count, 1)
 	}
+	
+	func testDCCAdmissionStateChangge_Then_flagIsSetInHealthCertifiedPerson() throws {
+		let vaccinationHealthCertificate: HealthCertificate = try vaccinationCertificate(type: .seriesCompletingOrBooster, ageInDays: 2)
+		let healthCertifiedPerson = HealthCertifiedPerson(
+			healthCertificates: [vaccinationHealthCertificate],
+			dccWalletInfo: DCCWalletInfo.fake(
+				admissionState: .fake(
+					visible: true,
+					badgeText: .fake(string: "3G"),
+					subtitleText: .fake(string: "3G")
+				)
+			)
+		)
+
+		let store = MockTestStore()
+		store.healthCertifiedPersons = [healthCertifiedPerson]
+
+		let newDCCWalletInfo: DCCWalletInfo = .fake(
+			admissionState: .fake(visible: true, badgeText: .fake(string: "2G+"))
+		)
+
+		var cclService = FakeCCLService()
+		cclService.dccWalletInfoResult = .success(newDCCWalletInfo)
+		cclService.didChange = true
+
+		let expectation = expectation(description: "dccWalletInfo updated")
+
+		let subscription = healthCertifiedPerson.$dccWalletInfo
+			.dropFirst()
+			.sink { _ in
+				if healthCertifiedPerson.isAdmissionStateChanged {
```

Comment:
I think I would add here an XCTAssertTrue to better show that this is our test core condition. Also for the future, it makes the test more stable.

Reply:
when the HealthCertficate service is initiated we can update wallet info twice the fist time the change is recorded but the second time there is no change so the flag is false and this causes the test to fail so in my implementation we can check that it was changed once at least